src_code/src/controllers/services/Authentication/SignUp.js

```diff
@@ -0,0 +1,163 @@
+/** Initialize neccessary modules */
+const bcrypt = require("bcryptjs");
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+const dbHelper = require("../../../utils/interfaces/IDBHelperFunctions");
+
+/**
+ * Handle user sign-up and insert the new account. The password is encrypted.
+ *
+ * @param {object} res - The response object for the HTTP request.
+ * @param {object} req - The request object from the HTTP request.
+ * @returns {object} - This response object indicates the result of the sign-up attempt.
+ * 
+ * Expected Request Body: 
+ * {
+ *      "userRole": "Student",
+ *      "firstName": "Le",
+ *      "middleName": "",
+ *      "lastName": "Tran",
+ *      "studentId": "", 
+ *      "emailAddress": "tran12312@spu.edu",
+ *      "accountPassword":"TallasGiraffe1!"
+ * }
+ */
+async function SignUp(res, req) {
+    try{ 
+        /** Validate if Sign Up information from request body is valid */
+        const errors = await SignUpValidation(res, req);
```

Comment:
nit: another redundant `await`

Reply:
Await is not redundant. It should be used with resolved Promise. Async function returns promise but javascript does not ensure that promist is successfully resolve. I will update it.